core/src/main/java/org/elasticsearch/snapshots/SnapshotsService.java

```diff
@@ -1116,6 +1180,18 @@ public static boolean isRepositoryInUse(ClusterState clusterState, String reposi
                 }
             }
         }
+        SnapshotDeletionsInProgress deletionsInProgress = clusterState.custom(SnapshotDeletionsInProgress.TYPE);
+        if (deletionsInProgress != null && deletionsInProgress.hasDeletionsInProgress()) {
+            return true;
+        }
+        RestoreInProgress restoreInProgress = clusterState.custom(RestoreInProgress.TYPE);
```

Comment:
you are right, I didn't notice the call to both in `RepositoriesService#ensureRepositoryNotInUse()`.  removing this.